presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,290 @@
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.DOUBLE, FieldType.FLOAT);
+        }
+        if (isVarcharType(columnType)) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+        }
+    }
+
+    private void checkFieldTypeOneOf(FieldType declaredFieldType, String columnName, Type columnType, FieldType... allowedFieldTypes)
+    {
+        if (!Arrays.asList(allowedFieldTypes).contains(declaredFieldType)) {
+            throw new IllegalArgumentException(format(
+                    "Wrong dataFormat '%s' specified for column '%s'; %s type implies use of %s",
+                    declaredFieldType.name(),
+                    columnName,
+                    columnType.getDisplayName(),
+                    Joiner.on("/").join(allowedFieldTypes)));
+        }
+    }
+
+    private int parseMapping(EncoderColumnHandle columnHandle, FieldType fieldType)
+    {
+        String mapping = Optional.ofNullable(columnHandle.getMapping()).orElse("0");
+        Matcher mappingMatcher = MAPPING_PATTERN.matcher(mapping);
+        if (!mappingMatcher.matches()) {
+            throw new IllegalArgumentException(format("invalid mapping format '%s' for column '%s'", mapping, columnHandle.getName()));
+        }
+        int start = parseInt(mappingMatcher.group(1));
+        OptionalInt end;
+        if (mappingMatcher.group(2) != null) {
+            end = OptionalInt.of(parseInt(mappingMatcher.group(2)));
+        }
+        else {
+            if (!isVarcharType(columnHandle.getType())) {
+                end = OptionalInt.of(start + fieldType.getSize());
+            }
+            else {
+                end = OptionalInt.empty();
+            }
+        }
+
+        checkArgument(start >= 0, "start offset %s for column '%s' must be greater or equal 0", start, columnHandle.getName());
+        end.ifPresent(endValue -> {
+            checkArgument(endValue >= 0, "end offset %s for column '%s' must be greater or equal 0", endValue, columnHandle.getName());
+            checkArgument(endValue > start, "end offset %s for column '%s' must greater than start offset", endValue, columnHandle.getName());
+        });
+
+        int length = end.isPresent() ? end.getAsInt() - start : fieldType.getSize();
+
+        if (!isVarcharType(columnHandle.getType())) {
+            checkArgument(!end.isPresent() || end.getAsInt() - start == length,
+                    "Bytes mapping for column '%s' does not match dataFormat '%s'; expected %s bytes but got %s",
+                    columnHandle.getName(),
+                    length,
+                    end.getAsInt() - start);
+        }
+
+        return length;
+    }
+
+    private static boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    private int getAndIncrementBufferPosition()
+    {
+        int currentPos = currentBufferPosition;
```

Comment:
nit: I think it would be more readable to separately get and increment in each `append*` method.